src/main/java/space/obminyashka/items_exchange/controller/AdvertisementController.java

```diff
@@ -112,6 +115,25 @@ public ResponseEntity<Page<AdvertisementTitleDto>> getPageOfAdvertisementsByKeyw
                 new ResponseEntity<>(allByKeyword, HttpStatus.OK);
     }
 
+    @GetMapping(ApiKey.ADV_SEARCH_PAGINATED_BY_CATEGORY_ID)
+    @Operation(summary = "Find advertisements by category id")
+    @ApiResponses(value = {
+            @ApiResponse(responseCode = "200", description = "OK"),
+            @ApiResponse(responseCode = "400", description = "BAD REQUEST"),
+            @ApiResponse(responseCode = "404", description = "NOT FOUND")})
+    public Page<AdvertisementTitleDto> getPageOfAdvertisementsByCategoryId(
+            @PathVariable("category_id") Long id,
+            @Parameter(name = "Results page you want to retrieve (0..N). Default value: 0")
+            @RequestParam(value = "page", required = false, defaultValue = "0") @PositiveOrZero int page,
+            @Parameter(name = "Number of records per page. Default value: 12")
+            @RequestParam(value = "size", required = false, defaultValue = "12") @PositiveOrZero int size) throws CategorySizeNotFoundException {
+        return Optional.of(advertisementService.findByCategoryId(id, PageRequest.of(page, size)))
+                .filter(Predicate.not(Page::isEmpty))
+                .orElseThrow(() -> new CategorySizeNotFoundException(
```

Comment:
I'd rather change that behavior. 
The exception should be thrown only in case when category ID does not exist in the DB otherwise the result should be returned with 200 response code even if that's empty.

Reply:
Done